### PR TITLE
Connect Tribe Commerce into the Share Capacity System

### PR DESCRIPTION
**Pull request by jesseeproductions**
_Ref:_ [C#10941 ](https://central.tri.be/issues/109419)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space between argument &quot;$global_stock&quot; and equals sign; 0 found | [Main.php:2578](https://github.com/moderntribe/event-tickets/blob/5eac6747e9e5b5c23c134cedc22bc335bba4c8c3/src/Tribe/Commerce/PayPal/Main.php#L2578)
:x: | Expected 1 space between argument &quot;$global_stock&quot; and equals sign; 0 found | [Main.php:2601](https://github.com/moderntribe/event-tickets/blob/5eac6747e9e5b5c23c134cedc22bc335bba4c8c3/src/Tribe/Commerce/PayPal/Main.php#L2601)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Commerce/PayPal/Main.php [hunk 1]
```diff
@@ -2560,32 +2566,66 @@ public function attendee_decreases_inventory( array $attendee ) {
...
-	 * @param int  $qty
+	 * @param int         $ticket_id       The ticket post ID
+	 * @param int         $qty             the quanitity to modify stock
+	 * @param bool        $shared_capacity true or false if the ticket is using share capacity
+	 * @param object|null $global_stock    the object of Tribe__Tickets__Global_Stock or null
 	 *
 	 * @return int
 	 */
-	public function increase_ticket_sales_by( $ticket_id, $qty = 1 ) {
+	public function increase_ticket_sales_by( $ticket_id, $qty = 1, $shared_capacity = false, $global_stock= null ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Main.php]:
```suggestion
	public function increase_ticket_sales_by( $ticket_id, $qty = 1, $shared_capacity = false, $global_stock = null ) {
```

File: src/Tribe/Commerce/PayPal/Main.php [hunk 2]
```diff
@@ -2560,32 +2566,66 @@ public function attendee_decreases_inventory( array $attendee ) {
...
-	 * @param int $qty
+	 * @param int         $ticket_id       The ticket post ID
+	 * @param int         $qty             the quanitity to modify stock
+	 * @param bool        $shared_capacity true or false if the ticket is using share capacity
+	 * @param object|null $global_stock    the object of Tribe__Tickets__Global_Stock or null
 	 *
 	 * @return int
 	 */
-	public function decrease_ticket_sales_by( $ticket_id, $qty = 1 ) {
+	public function decrease_ticket_sales_by( $ticket_id, $qty = 1, $shared_capacity = false, $global_stock= null ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Main.php]:
```suggestion
	public function decrease_ticket_sales_by( $ticket_id, $qty = 1, $shared_capacity = false, $global_stock = null ) {
```

File: src/Tribe/Commerce/PayPal/Main.php [hunk 3]
```diff
@@ -2560,32 +2566,66 @@ public function attendee_decreases_inventory( array $attendee ) {
...
+	 * @param object|null $global_stock    the object of Tribe__Tickets__Global_Stock or null
 	 *
 	 * @return int
 	 */
-	public function increase_ticket_sales_by( $ticket_id, $qty = 1 ) {
+	public function increase_ticket_sales_by( $ticket_id, $qty = 1, $shared_capacity = false, $global_stock= null ) {
 		$sales = (int) get_post_meta( $ticket_id, 'total_sales', true );
 		update_post_meta( $ticket_id, 'total_sales', $sales + $qty );
 
+		if ( $shared_capacity ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Main.php]:
```suggestion
		if ( $shared_capacity && $global_stock instanceof Tribe__Tickets__Global_Stock ) {
```

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space between argument &quot;$global_stock&quot; and equals sign; 0 found | [Main.php:2578](https://github.com/moderntribe/event-tickets/blob/5e9929fd9114e92744c43a34300dcb807f3de283/src/Tribe/Commerce/PayPal/Main.php#L2578)
:x: | Expected 1 space between argument &quot;$global_stock&quot; and equals sign; 0 found | [Main.php:2601](https://github.com/moderntribe/event-tickets/blob/5e9929fd9114e92744c43a34300dcb807f3de283/src/Tribe/Commerce/PayPal/Main.php#L2601)

> via jenkins/codesniffer-phpcs

**Comment by sc0ttkclark** [comment]
Test for stock not passing, go ahead and check that when you have time @jesseeproductions https://travis-ci.org/moderntribe/event-tickets/builds/501591515?utm_source=github_status&utm_medium=notification

**Pull request merged**